src/main/java/com/amazonaws/encryptionsdk/CreateDecryptingInputStreamRequest.java

```diff
@@ -62,6 +83,39 @@ public Builder inputStream(InputStream inputStream) {
             return this;
         }
 
+        /**
+         * Sets the maxBodySize
+         *
+         * @param maxBodySize The maximum length of encrypted content that is
+         *   allowed to be decrypted at once. Decryption will fail on any encrypted message
+         *   that requires performing one decryption operation on a length greater
+         *   than this value.
+         * @return The Builder, for method chaining
+         */
+        public Builder maxBodySize(Integer maxBodySize) {
+            if (maxBodySize < 0) {
+                throw new IllegalArgumentException("maxBodySize must be null or non-negative");
+            }
+            this.maxBodySize = maxBodySize;
+            return this;
+        }
+
+        /**
+         * Sets the maxHeaderSize
+         *
+         * @param maxHeaderSize The maximum length of message header that is
+         *   allowed to be deserialized. Decryption will fail on any encrypted message
+         *   with a message header length greater than this value.
+         * @return The Builder, for method chaining
+         */
+        public Builder maxHeaderSize(Integer maxHeaderSize) {
+            if (maxHeaderSize < 0) {
+                throw new IllegalArgumentException("maxHeaderSize must be null or positive");
```

Comment:
Same here, should this be `<= 0 ` or should the `IllegalArgumentException` be updated?